src/lava/magma/compiler/compiler.py

```diff
@@ -271,16 +303,42 @@ def _compile_proc_models(
                     # and Ports
                     v = [VarInitializer(v.name, v.shape, v.init, v.id)
                          for v in p.vars]
-                    ports = (list(p.in_ports) + list(p.out_ports)
-                             + list(p.ref_ports))
+                    ports = (list(p.in_ports) + list(p.out_ports))
                     ports = [PortInitializer(pt.name,
                                              pt.shape,
-                                             getattr(pm, pt.name).d_type,
+                                             self._get_port_dtype(pt, pm),
                                              pt.__class__.__name__,
                                              pp_ch_size) for pt in ports]
+                    # Create RefPort (also use PortInitializers)
+                    ref_ports = list(p.ref_ports)
+                    ref_ports = [
+                        PortInitializer(pt.name,
+                                        pt.shape,
+                                        self._get_port_dtype(pt, pm),
+                                        pt.__class__.__name__,
+                                        pp_ch_size) for pt in ref_ports]
+                    # Create VarPortInitializers (contain also the Var name)
+                    var_ports = []
+                    for pt in list(p.var_ports):
+                        var_ports.append(
+                            VarPortInitializer(
+                                pt.name,
+                                pt.shape,
+                                pt.var.name,
+                                self._get_port_dtype(pt, pm),
+                                pt.__class__.__name__,
+                                pp_ch_size,
+                                self._map_var_port_class(pt, proc_groups)))
+
+                        # Set implicit VarPorts as attribute to ProcessModel
```

Comment:
Is it explained anywhere what ImplicitVarPorts are and why we create them?

Reply:
Done